File: app/src/main/java/com/jeksonshar/jweather/ui/WeatherViewHolder.java
package com.jeksonshar.jweather.ui;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.jeksonshar.jweather.R;
import com.jeksonshar.jweather.model.WeatherModel;
import com.squareup.picasso.Picasso;

import java.text.DecimalFormat;
import java.text.SimpleDateFormat;
import java.util.Locale;

public class WeatherViewHolder extends RecyclerView.ViewHolder {

    private final TextView date;
    private final ImageView weatherStateImage;
    private final TextView temp;
    private final TextView windSpeed;
    private final TextView airPressure;
    private final TextView humidity;
    private final TextView weatherState;

    private WeatherModel currantItemWeather;

    public WeatherViewHolder(@NonNull View itemView) {
        super(itemView);

        date = itemView.findViewById(R.id.date_item_weather);
        weatherStateImage = itemView.findViewById(R.id.view_weather_state);
        temp = itemView.findViewById(R.id.max_min_temp);
        windSpeed = itemView.findViewById(R.id.wind_speed);
        airPressure = itemView.findViewById(R.id.air_pressure);
        humidity = itemView.findViewById(R.id.humidity);
        weatherState = itemView.findViewById(R.id.weather_state);
    }

    public void bindTo(WeatherModel weatherModel) {

        SimpleDateFormat dateFormat = new SimpleDateFormat(
                "EEEE\nd MMM yyyy", Locale.getDefault());
        date.setText(dateFormat.format(weatherModel.getDate()));

        Picasso.get()
                .load("https://www.metaweather.com/static/img/weather/png/" + weatherModel.getWeatherStateAbbr() + ".png")
                .into(weatherStateImage);

        String minMaxTemp = Math.round(weatherModel.getTempMax()) + " / " + Math.round(weatherModel.getTempMin()) + " ˚C";
        temp.setText(minMaxTemp);

        float wind = weatherModel.getWindSpeed();
        DecimalFormat decimalFormat = new DecimalFormat("#.#");
        String windSpeed = decimalFormat.format(wind*0.44704) + " м/с"; // convert from mph to m/s
        this.windSpeed.setText(windSpeed);

        float pressure = weatherModel.getAirPressure();
        String airPressure = Math.round(pressure/1.33322) + " мм"; // convert from mBar to mmHg
        this.airPressure.setText(airPressure);

        String humidity = Math.round(weatherModel.getHumidity()) + " %";
        this.humidity.setText(humidity);

        weatherState.setText(weatherModel.getWeatherState());
    }
}

File: app/src/main/java/com/jeksonshar/jweather/ui/WeatherViewHolder.kt
package com.jeksonshar.jweather.ui

import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.jeksonshar.jweather.R
import com.jeksonshar.jweather.model.WeatherModel
import com.squareup.picasso.Picasso
import java.text.DecimalFormat
import java.text.SimpleDateFormat
import java.util.*
import kotlin.math.roundToInt
//TODO между переменными и методами лучше оставлять пробелы, лечге читается
class WeatherViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private val date: TextView = itemView.findViewById(R.id.date_item_weather)
    private val weatherStateImage: ImageView = itemView.findViewById(R.id.view_weather_state)
    private val temp: TextView = itemView.findViewById(R.id.max_min_temp)
    private val windSpeed: TextView = itemView.findViewById(R.id.wind_speed)
    private val airPressure: TextView = itemView.findViewById(R.id.air_pressure)
    private val humidity: TextView = itemView.findViewById(R.id.humidity)
    private val weatherState: TextView = itemView.findViewById(R.id.weather_state)
    fun bindTo(weatherModel: WeatherModel) {
        val dateFormat = SimpleDateFormat(
                "EEEE\nd MMM yyyy", Locale.getDefault()) //TODO можно вынести в глобальную переменную
        date.text = dateFormat.format(weatherModel.date!!)
        Picasso.get()
                .load("https://www.metaweather.com/static/img/weather/png/" + weatherModel.weatherStateAbbr + ".png") //TODO Это лучше формировать во ViewModel, сюда кидать готовую строчку
                .into(weatherStateImage)
        val minMaxTemp = weatherModel.tempMax.roundToInt().toString() + " / " + weatherModel.tempMin.roundToInt() + " ˚C" //TODO Это лучше формировать во ViewModel, сюда кидать готовую строчку
        temp.text = minMaxTemp
        val wind = weatherModel.windSpeed
        val decimalFormat = DecimalFormat("#.#")
        val windSpeed = decimalFormat.format(wind * 0.44704) + " м/с" // convert from mph to m/s //TODO Это лучше формировать во ViewModel, сюда кидать готовую строчку
        this.windSpeed.text = windSpeed
        val pressure = weatherModel.airPressure
        val airPressure = (pressure / 1.33322).roundToInt().toString() + " мм" // convert from mBar to mmHg //TODO Это лучше формировать во ViewModel, сюда кидать готовую строчку
        this.airPressure.text = airPressure
        val humidity = weatherModel.humidity.roundToInt().toString() + " %" //TODO Это лучше формировать во ViewModel, сюда кидать готовую строчку
        this.humidity.text = humidity
        weatherState.text = weatherModel.weatherState
    }

}